spec/requests/api/vms_spec.rb

```diff
@@ -1431,4 +1431,99 @@ def update_raw_power_state(state, *vms)
+        :owner    => vm
+      )
+      api_basic_authorize(action_identifier(:vms, :read, :resource_actions, :get))
+
+      run_get(vms_url(vm.id))
+
+      expected = {
+        "actions" => a_collection_including(
+          a_hash_including("name" => "button1"),
+          a_hash_including("name" => "button2")
+        )
+      }
+      expect(response.parsed_body).to include(expected)
+    end
+
+    it "renders the custom actions when requested" do
+      vm = FactoryGirl.create(:vm_vmware)
+      FactoryGirl.create(:custom_button, :name => "button1", :applies_to => vm)
+      FactoryGirl.create(
+        :custom_button_set,
+        :name     => "button_group1",
+        :children => [FactoryGirl.create(:custom_button, :name => "button2", :applies_to => vm)],
+        :owner    => vm
+      )
+      api_basic_authorize(action_identifier(:vms, :read, :resource_actions, :get))
+
+      run_get(vms_url(vm.id), :attributes => "custom_actions")
+
+      expected = {
+        "custom_actions" => a_hash_including(
+          "buttons"       => [
+            a_hash_including("name" => "button1")
+          ],
+          "button_groups" => [
+            a_hash_including(
+              "name"    => "button_group1",
+              "buttons" => [
+                a_hash_including("name" => "button2")
+              ]
+            )
+          ]
+        )
+      }
+      expect(response.parsed_body).to include(expected)
+    end
+
+    it "renders the custom action buttons when requested" do
+      vm = FactoryGirl.create(:vm_vmware)
+      FactoryGirl.create(:custom_button, :name => "button1", :applies_to => vm)
+      FactoryGirl.create(
+        :custom_button_set,
+        :children => [FactoryGirl.create(:custom_button, :name => "button2", :applies_to => vm)],
+        :owner    => vm
+      )
+      api_basic_authorize(action_identifier(:vms, :read, :resource_actions, :get))
+
+      run_get(vms_url(vm.id), :attributes => "custom_action_buttons")
+
+      expected = {
+        "custom_action_buttons" => a_collection_containing_exactly(
+          a_hash_including("name" => "button1"),
+          a_hash_including("name" => "button2")
+        )
+      }
+      expect(response.parsed_body).to include(expected)
+    end
+
+    it "can execute a custom action" do
+      vm = FactoryGirl.create(:vm_vmware)
+      FactoryGirl.create(
+        :custom_button,
+        :name            => "test_button",
+        :applies_to      => vm,
+        :resource_action => FactoryGirl.create(:resource_action)
+      )
+      api_basic_authorize
+
+      run_post(vms_url(vm.id), :action => "test_button", :button_key1 => "foo")
+
+      expected = {
+        "success" => true,
+        "message" => "Invoked custom action test_button for vms id: #{vm.id}",
+        "href"    => a_string_matching(vms_url(vm.id))
+      }
+      expect(response.parsed_body).to include(expected)
+    end
+  end
```

Comment:
I might, but prefer to do that in a future PR. The current specs there are set up assuming that the service template is at the center of what's being tested, so would at least need to scope the existing tests in a new context. The resultant diff I thought would be unnecessarily large for this PR